### PR TITLE
(fix): refactor login logic from js to ts

### PR DESCRIPTION
**Pull request by robpan38**

### CODE REVIEW

**Comment by robpan38** [comment]
[refactor js to ts](https://trello.com/c/ChZ1Exaf/8-refactor-js-to-ts)

File: package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "dependencies": {
-    "express": "^4.18.1"
+    "express": "^4.18.1",
+    "typescript": "^4.7.4"
```
Comment (alexdudescu) [inline package.json]: Should typescript be a dependency or a dev dependency?

File: scripts/loginLogic.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-});
-
-loginBtn.addEventListener("click", () => {
-    if (loginBtn.classList.contains("active-login-btn")) {
-        loginBtn.classList.remove("active-login-btn");
-        loginInput.value = "";
-        location.href = "./chat.html";
-    }
-})
+(function () {
```
Comment (alexdudescu) [inline scripts/loginLogic.js]: Shouldn't we kill this if we migrate to ts?

File: package.json (hunk 1 above)
Comment (robpan38) [reply package.json]: fixed in the last commit

File: scripts/loginLogic.js (hunk 2 above)
Comment (robpan38) [reply scripts/loginLogic.js]: fixed in the last commit

**Review by mihneat — APPROVED** [review]

**Pull request merged**